geonode_mapstore_client/client/js/epics/gnresource.js

```diff
@@ -130,14 +131,15 @@ const resourceTypes = {
                         ? new Promise(resolve => resolve(options.resourceData))
                         : isDefaultDatasetSubtype(subtype)
                             ? getDatasetByPk(pk)
-                            : getResourceByPk(pk)
+                            : getResourceByPk(pk),
+                    getDatasetTimeSettingsByPk(pk)
```

Comment:
Is it not possible to include timeseries in the main getDatasetByPk request? (like we have include[]=data for maps?)
It would be easier to have a single request.
Note: If we need to make another request we need to perform it only for dataset that supports it so probably we need to do the dataset request and then based on the response add the new one (eg 3D Tiles does not support it and at the moment they uses the resources endpoint )

Reply:
Updated